apps/backend/src/lib/controller/notification/telegram/client.ts

```diff
@@ -0,0 +1,114 @@
+import { Bot } from "grammy";
+import { iNotificationClient } from "../interfaces";
+import { PrismaClient } from "@prisma/client";
+
+export class TelegramNotificationClient implements iNotificationClient {
+  private bot: Bot | undefined;
+  private readonly maxRetries = 3;
+  private readonly retryDelay = 5000;
+  prismaClient: PrismaClient | undefined;
+
+  constructor() {
+    if (!process.env.TELEGRAM_BOT_API) {
+      // No token, so we don't initialize
+      return;
+    }
+    this.bot = new Bot(process.env.TELEGRAM_BOT_API);
+    this.prismaClient = new PrismaClient();
+  }
+
+  private async setupBot(): Promise<void> {
+    if (!this.bot) {
+      throw new Error("Bot is not initialized");
+    }
+
+    this.bot.command("start", async (ctx) => {
+      // The start parameter will be available in ctx.match
+      const startParameter = ctx.match;
+
+      if (startParameter) {
+        // This is where you can link the Telegram user with your backend user
+        const telegramUserId = ctx.from?.id;
+        const backendUserId = startParameter;
+
+        if (!telegramUserId || !backendUserId) {
+          throw new Error("Invalid user IDs");
+        }
+
+        try {
+          // Store the association in your database
+          await this.prismaClient?.user.update({
+            where: { id: backendUserId },
+            data: { notificationUserId: telegramUserId?.toString() },
+          });
+
+          await ctx.reply(
+            `I'm Curtis, the Cursive Connections bot. I'm here to help you discover and deepen your connections.`
+          );
+        } catch (error) {
+          await ctx.reply(
+            "Sorry, there was an error linking your account. Please try connecting again."
+          );
+          console.error("Linking error:", error);
+        }
+      } else {
+        await ctx.reply(
+          "Welcome! Please register for notifications from your profile page."
+        );
+      }
+    });
+
+    this.bot.start({
+      onStart: (info) => {
+        console.log(`[TELEGRAM] Started bot '${info.username}' successfully!`);
+      },
+    });
+  }
+
+  async Initialize(): Promise<void> {
+    const startDelay = parseInt(process.env.TELEGRAM_BOT_START_DELAY_MS ?? "0");
+    await new Promise((resolve) => setTimeout(resolve, startDelay));
+
+    for (let i = 0; i < this.maxRetries; i++) {
+      try {
+        await this.setupBot();
+        console.log("Telegram notification client initialized");
+        return;
+      } catch (error) {
+        console.log(
+          `Telegram initialization failed, retrying in ${this.retryDelay}ms...`
+        );
+        await new Promise((resolve) => setTimeout(resolve, this.retryDelay));
```

Comment:
<img width="682" alt="image" src="https://github.com/user-attachments/assets/535df312-d327-4ca4-ac72-26c0062c204d">

I intentionally did not set the env var to test this.  Will update the log so that it's explicit that we exited without initializing the bot.

Reply:
Updated:

<img width="679" alt="image" src="https://github.com/user-attachments/assets/659a67a0-3e77-46b6-b1a6-b4d67c254dea">

Reply:
Addressed: https://github.com/cursive-team/connections/pull/154/commits/7ce98da4ba460144b48f1f3580d3490aed651771

Reply:
Previously after the max retries it would silently exit the method and enter the `.then` case in the api handler, which prints `Notification service initialized`, despite being not being properly set up.